Adjust other support pages (#153)
hibernate/hibernate.org: search/support.adoc

[[supported-versions]]
== Supported Products pass:[<i class="icon-user-md icon-fixed-width icon-2x"></i>]

> Hibernate Search is a fully supported component of the https://www.redhat.com/en/products/runtimes[Red Hat Runtimes] offering.
> 

Reviewer: ```suggestion
Hibernate Search is a fully supported component of Red Hat JBoss EAP as part of the https://www.redhat.com/en/products/runtimes[Red Hat Runtimes] offering.
```

Using JBoss EAP, Quarkus, Open Liberty or Spring Boot? They are all part of Red Hat Runtimes.

> In addition, the https://www.redhat.com/en/technologies/jboss-middleware/data-grid[Red Hat Data Grid] provides optional extensions such as high performance real time replication for Hibernate Search powered applications running on one of the Red Hat supported runtime.
> 

Reviewer: I'd remove this. We're removing the Infinispan directory provider in Hibernate Search 6, and Infinispan dropped it in version 11, so RHDG might follow pretty soon.

```suggestion
```

> The Red Hat Data Grid also embeds a custom edition of Hibernate Search and allows using most of the same API to run high performance full-text queries and data mining on the data grid.

Author: Not totally happy of how this turned out. @yrodiere could you have a look and adjust as you see fit?

Reviewer: They are in the process of removing Hibernate Search APIs from Infinispan, so I'd go with this:

```suggestion
[Red Hat Data Grid](https://www.redhat.com/en/technologies/jboss-middleware/data-grid) also uses a custom edition of Hibernate Search internally to provide high performance full-text queries and data mining on the data grid.
```
